frontend/src/metabase/dashboard/actions/tabs.ts

```diff
@@ -109,6 +110,15 @@ export function getDefaultTab({
   };
 }
 
+export function getIdFromSlug(slug: string | undefined) {
```

Comment:
This can be useful in other places since we use slugs for many entities, it worth moving to shared utils

Reply:
It looks like there is already an `extractEntityId` function in `metabase/lib/utils/urls/utils.ts` that's used extensively throughout the codebase (46 references).

They effectively do the same thing, but I'm a bit hesitant to replace `getIdFromSlug` with it, because the types and edge case behavior defined in the unit test are a bit difference.